final-project/application.py

```diff
@@ -516,33 +516,19 @@ def product():
         return apology("Sorry, we didn't find a product with id "
             + id_products_request)
 
-    # Return JSON with product info
-    return jsonify(product_info[0])
-
-    # Assign local variables
-    product_name = product_info[0]["product_name"]
-    image = product_info[0]["image"]
-    link = product_info[0]["link"]
-    description = product_info[0]["description"]
-    brand = product_info[0]["brand"]
-    price = usd(product_info[0]["price"])
 
-    # Get product references
+    # Get product references for a given product name.
     (number_of_references,
     reference_titles,
-    reference_links) = get_reference(product_name)
-
-    #Render the product page.
-    return render_template("product.html",
-                            product_name=product_name,
-                            image=image,
-                            link=link,
-                            description=description,
-                            brand=brand,
-                            price=price,
-                            number_of_references=number_of_references,
-                            reference_titles=reference_titles,
-                            reference_links=reference_links)
+    reference_links) = get_reference(product_info[0]["product_name"])
+
+    # add references to our product information
+    product_info[0]['number_of_references'] = number_of_references
+    product_info[0]['reference_titles'] = reference_titles
+    product_info[0]['reference_links'] = reference_links
+
+    # Return json of product information
+    return jsonify(product_info[0])
```

Comment:
Why only attach to the first element? If this is one thing, it shouldn't need to be in an array.

Reply:
It's the .execute function I use. It stores the result of SELECT queries in a list of dict objects. Unless I switch to a different library, then I have to access the [0] element. I do realize the inefficiency of this though. 

Similar to the comment in the last PR, if I switch to PostgreSQL then it would be a good time to address db optimization.

Reply:
There is a much better and simpler way than that. If you go to here:

https://github.com/jwq1/CS50x_jwq1/pull/43/files/5aafb8c3b2233e2e4934a749506a1eb754d44238#diff-ef5e1ecce45f4790e39e7d974edd28d6R515

```
product_info = find_product(id_products_request)[0]
```

And an even better refactor is to return only one element from `find_product`. In fact, this may not even be the right method. Your method should be:

```python
# @param id [String] ID of product
#
# @returns [Product] A single product
def get_product(id):
  # ...
```